trillian/migrillian/core/controller.go

```diff
@@ -30,9 +31,41 @@ import (
 
 	"github.com/google/trillian/merkle"
 	_ "github.com/google/trillian/merkle/rfc6962" // Register hasher.
+	"github.com/google/trillian/monitoring"
 	"github.com/google/trillian/types"
 )
 
+var (
+	metrics     treeMetrics
+	metricsOnce sync.Once
+)
+
+// treeMetrics holds metrics keyed by Tree ID.
+type treeMetrics struct {
+	masterRuns     monitoring.Counter
+	masterCancels  monitoring.Counter
+	isMaster       monitoring.Gauge
+	entriesFetched monitoring.Counter
+	entriesSeen    monitoring.Counter
+	entriesStored  monitoring.Counter
+	// TODO(pavelkalinnikov): Add latency histograms, latest STH, tree size, etc.
+}
+
+// initMetrics creates metrics using the factory, if not yet created.
+func initMetrics(mf monitoring.MetricFactory) {
+	const treeID = "treeID"
```

Comment:
Existing metrics use either a label of `"logid"` or `"tree_id"` -- best not to introduce a third variant.

Reply:
Done.